Liquidity-provider-server automated update of documentation update-from-lps-20260224-213714 (#472)
rsksmart/devportal: rif/identity/contribute.md

- Pull requests should point to `develop` by default - other cases are explained bellow

Thanks for collaborating and contributing!

## Main branches 

All the code that is in `master` (or `main`) is code that is productive code: it means it is deployed or published as an NPM package. In our case, sometimes the release gets delayed due to some external dependencies, so we can assume that if the code is in `master`, it is ready to be delivered.
The code present in `develop` is part of the current work and, in most of the cases, is the immediate release, so it may be deployed or published as a _beta_ version. Neither `master` nor `develop` are personal branches, they are public branches and the code present there should compile and be ready to use/test. These branches are protected and are updated only through PRs.

`master` is updated through PRs from `develop` (normal case), `hotfix` branches or `release` branches. That PR should be merged using the _“Create a merge commit”_ option. We want to also identify the new versions with merge commits. After the PR is merged and everything deployed, a release tag is created.

> IMPORTANT
> Every pull request to `develop` should be merged through a REBASE or a SQUASH commit.

Reviewer: ```suggestion
**Important**

Every pull request to `develop` should be merged through a **rebase** or a **squash** commit.
```

> Every pull request to `master` should be merged through a MERGE commit.

Reviewer: ```suggestion
Every pull request to `master` should be merged through a **merge** commit.
```

If the blocking stuff is code related and we are ready to address that, then a new `feature` branch will be created, but in this case, instead of creating it from `develop`, we create it from the `release` branch we’ve created before. This code is going to be merged into the `release` branch using the same strategy we used to merge into `develop`.
Once the `release` branch is ready, a PR is created to merge that code into `master` and then be released, but it should be also merged into `develop` because there could be code that has been added after the `release` branch has been created. 

Depending on the case, the PR to merge back into `develop` could be simple or not, sometimes it can throw unexpected conflicts, in those cases, the best option could be an auxiliary branch and use the cherry-pick command in order to pick the desired commits, and then create a PR to `develop` from that auxiliary branch.

> IMPORTANT
> Every pull request to a `release` branch should be merged through _“Rebase and merge”_ or _“Squash and merge”_ options.

Reviewer: ```suggestion
**Important**

Every pull request to a `release` branch should be merged through _“Rebase and merge”_ or _“Squash and merge”_ options.
```

---
layout: rsk
---

# RIF Identity Git approach

> 

Reviewer 2: ```suggestion

**Quick summary**

- `master` has latest release - deployment or package distribution
- `develop` has latest approved feature - feature was reviewed and approved, but not yet released as final
- Pull requests should point to `develop` by default - other cases are explained bellow

Thanks for collaborating and contributing!

```

Strongly based on the [branching model](https://nvie.com/posts/a-successful-git-branching-model/) proposed by Vincent Driessen. Also named Gitflow in [this article](https://www.atlassian.com/git/tutorials/comparing-workflows/gitflow-workflow).

It is a strict branching model that refers to two main branches: `master` (or _main_) and `develop` (or _dev_).

## Main branches 

> All the code that is in `master` is code that is productive code: it means it is deployed or pubished as an NPM package. In our case, sometimes the release gets delayed due to some external dependencies, so we can assume that if the code is in `master`, it is ready to be delivered.

Reviewer 2: ```suggestion
All the code that is in `master` (or `main`) is code that is productive code: it means it is deployed or published as an NPM package. In our case, sometimes the release gets delayed due to some external dependencies, so we can assume that if the code is in `master`, it is ready to be delivered.
```
